github_connector_odoo/models/odoo_module.py

```diff
@@ -178,3 +178,23 @@ def create_if_not_exist(self, technical_name):
 
     def name_get(self):
         return [(module.id, module.technical_name) for module in self]
+
+    def action_odoo_module_version(self):
+        self.ensure_one()
+        action = self.env.ref(
+            "github_connector_odoo.action_odoo_module_version"
+        ).read()[0]
+        action["context"] = dict(self.env.context)
+        action["context"].pop("group_by", None)
+        action["context"]["search_default_module_id"] = self.id
+        return action
+
+    def action_odoo_module_version_dependency(self):
+        self.ensure_one()
+        action = self.env.ref(
+            "github_connector_odoo.action_odoo_module_version"
+        ).read()[0]
+        action["context"] = dict(self.env.context)
+        action["context"].pop("group_by", None)
```

Comment:
Both the `action_odoo_module_version()` and `action_odoo_module_version_dependency()` functions share the same logic, with exception for the final `search_*` assignment. Would it be possible to move the shared logic into a 3rd function that both of them call?

Reply:
I was thinking more in the following: having a mixin that add a generic method `action_open` that passes 3 arguments:

- XML-ID of the action to read.
- Name of the field to look for (then you add later the `search_default` prefix).
- ID of the record for that field.

Put the logic that way.

Then inherit all models from that mixin and call the method from each button with the proper arguments.

Reply:
Yes, inheritance seems like a better option, indeed. I just didn't know if we could share the logic between all the models, but would certainly make it a lot more maintainable :+1: .

Reply:
OK, thanks, good suggestion. I work in this change.